quantestpy/simulator/pauli_circuit.py

```diff
@@ -131,8 +118,26 @@ def _execute_i_th_gate(self, i: int) -> None:
             elif gate["name"] == "swap":
                 self._execute_swap_gate(gate["target_qubit"])
             else:
-                raise QuantestPyError("Unexpected error. Please report.")
+                raise PauliCircuitError(
+                    "Unexpected error. Please report."
+                )
 
     def _execute_all_gates(self,) -> None:
         for i in range(len(self._gates)):
             self._execute_i_th_gate(i)
+
+
+def cvt_quantestpy_circuit_to_pauli_circuit(
```

Comment:
This is the converter from `QuantestPyCircuit` to `PauliCircuit`. This will be called in the assert methods which assume `PauliCircuit` internally.